zkevm-circuits/src/state_circuit/lexicographic_ordering.rs

```diff
@@ -0,0 +1,278 @@
+use super::{N_LIMBS_ACCOUNT_ADDRESS, N_LIMBS_ID, N_LIMBS_RW_COUNTER};
+use crate::{
+    evm_circuit::{param::N_BYTES_WORD, witness::Rw},
+    gadget::is_zero::{IsZeroChip, IsZeroConfig, IsZeroInstruction},
+    util::Expr,
+};
+use eth_types::{Field, ToBigEndian};
+use halo2_proofs::{
+    circuit::Region,
+    plonk::{Advice, Column, ConstraintSystem, Error, Expression, Fixed, VirtualCells},
+    poly::Rotation,
+};
+use itertools::Itertools;
+use std::ops::Mul;
+
+// 2 limbs for tag, field_tag and id.
+// 10 limbs for address,
+// 16 limbs for storage key
+// 2 limbs for rw_counter
+// 30 limbs in total -> can fit into two field elements
+#[derive(Clone)]
+pub struct Config<F: Field> {
+    diff_1: Column<Advice>,
```

Comment:
Would it be possible to add a short description of this chip and especially of `diff_1` and `diff_2` fields?

Reply:
I've added a comment describing this chip. I would love to hear any suggestions for better names for `diff_1` and `diff_2`.

Reply:
Thanks, it makes reading the code much easier! For `diff_1` and `diff_2` naming I am not sure - perhaps something that would give some intuition, like `aggregated_limbs_1`, `aggregated_limbs_2`? It would be also good perhaps to add some rationale why limbs are split into two parts, it's because of field overflow, right?

Reply:
Why not just encode 6 fields into two 240-bit values, `A_hi` and `A_lo`? Then you just need to compare two 240-bit values, and only the diff needs to be encoded in 16-bit limbs for range check.

Reply:
That would definately be easier to understand and explain, but it would require 30 advice columns for the limbs of the `diff_hi` and `diff_lo` instead of the 4 we use now.

Reply:
agree.  Proving cost of adding new gates is much smaller than adding new columns.